### PR TITLE
Add openssl/gnutls tls1.3 feature tests.

### PR DESCRIPTION
**Pull request by yuhaoth**
## Description
Add functions and test cases to make sure tls1.3 is available in openssl/gnutls.
Relative issues: #4417 

## Status
**IN DEVELOPMENT**

## Requires Backporting
NO  

## Migrations
NO

## Todos
- [x] Tests
- [ ] Documentation
- [ ] Changelog updated
- [ ] Backported

## Steps to test or reproduce
Outline the steps to test or reproduce the PR here.

### CODE REVIEW

File: tests/ssl-opt.sh [hunk 1]
```diff
@@ -8478,6 +8508,23 @@ run_test    "export keys functionality" \
...
+# openssl/gnutls feature tests: check if tls1.3 exists.
+requires_openssl_tls1_3
+run_test    "TLS1.3: Test openssl tls1_3 feature" \
+            "$O_SRV -tls1_3 -msg" \
+            "$O_CLI -tls1_3 -msg" \
+            0 \
+            -c "TLS 1.3" \
+            -s "TLS 1.3"
+
+requires_openssl_tls1_3
```
Comment (hanno-arm) [inline tests/ssl-opt.sh:8520]: This should be gnutls?

File: tests/ssl-opt.sh [hunk 2]
```diff
@@ -346,6 +346,56 @@ requires_openssl_legacy() {
...
+    fi
+}
+# skip next test if tls1_3 is not available
+requires_openssl_tls1_3() {
+    requires_openssl
+    if [ "$OPENSSL_AVAILABLE" = "NO" ]; then
+        OPENSSL_TLS1_3_AVAILABLE="NO"
+    fi
+    if [ -z "${OPENSSL_TLS1_3_AVAILABLE:-}" ]; then
+        if $OPENSSL_CMD s_client -help 2>&1 | grep tls1_3 >/dev/null
```
Comment (mpg) [inline tests/ssl-opt.sh:368]: Hmm, the problem with that is the currently on the CI the default OpenSSL is 1.0.2g which doesn't have support for TLS 1.3, so the corresponding tests are always going to be skipped in the CI; we should use `$OPENSSL_NEXT` instead.

Apparently, this script is not using `OPENSSL_NEXT` yet, but you should be able to add support based on what's done for `GNUTLS_NEXT`and `OPENSSL_LEGACY`: set the variables `$O_NEXT_CLI` and `$O_NEXT_SRV` where similar variables are set for legacy, then change the previous function from `requires_openssl` to `requires_openssl_next` and using that here.

**Review by mpg — CHANGES_REQUESTED** [review]
Looks pretty good to me, except we should use `OPENSSL_NEXT` in order for the test to be run, rather than skipped, in the CI.

**Review by mpg — COMMENTED** [review]
Looks good to me, let's for until the CI completes its run and check in the logs to confirm the test is no longer skipped.

**Review by mpg — APPROVED** [review]
I was able to confirm from the CI logs that both tests are now run on the CI, so it's all good!

**Review by hanno-arm — APPROVED** [review]
LGTM too

**Pull request merged**